packages/office-ui-fabric-react/src/utilities/selection/SelectionZone.tsx

```diff
@@ -139,6 +139,8 @@ export class SelectionZone extends BaseComponent<ISelectionZoneProps, {}> {
         } else {
           this._onItemSurfaceClick(ev, index);
         }
+      } else {
+        selection.setAllSelected(false);
```

Comment:
Without this line, if you up into a header, the selection stays on the previous row. But its better if you're doing multi select. 
![groupdetailslist2](https://cloud.githubusercontent.com/assets/7392105/26381458/d909bca4-3fd9-11e7-8101-fa8d0bca5fdc.gif)